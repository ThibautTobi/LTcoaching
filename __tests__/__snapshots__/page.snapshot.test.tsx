import { render } from '@testing-library/react';
import Home from '../../app/page';
import React from 'react';
import '@testing-library/jest-dom';

// --- Mocks des dépendances externes ---

// Mock de next/image
jest.mock('next/image', () => {
  // eslint-disable-next-line react/display-name
  const MockedImage = ({
    alt,
    ...rest
  }: React.ImgHTMLAttributes<HTMLImageElement>) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img alt={alt || ''} {...rest} />;
  };
  return { __esModule: true, default: MockedImage };
});

// Mock de next/link
jest.mock('next/link', () => {
  // eslint-disable-next-line react/display-name
  const MockedLink = ({
    children,
    href,
    ...rest
  }: {
    children: React.ReactNode;
    href: string;
  }) => (
    <a href={href} {...rest}>
      {children}
    </a>
  );

  return { __esModule: true, default: MockedLink };
});

// --- Groupe de tests ---
describe('Home page snapshot tests', () => {
  it('devrait correspondre à un snapshot stable', () => {
    const { asFragment } = render(<Home />);

    // Vérifie que le rendu de la page correspond au snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
