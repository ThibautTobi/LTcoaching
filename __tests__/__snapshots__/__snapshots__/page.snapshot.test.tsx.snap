// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home component snapshot tests correspond au snapshot 1`] = `
<DocumentFragment>
  <main
    class="min-h-screen bg-background text-foreground"
  >

    <div
      class="relative w-full h-[250px] md:h-[300x] lg:h-[350px] xl:h-[450px] 2xl:h-[500px] overflow-hidden"
    >
      <img
        alt="image-sport"
        class="object-cover w-full h-full"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes=" 100vw "
        src="/_next/image?url=%2Fsport-acceuil.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fsport-acceuil.jpg&w=640&q=75 640w, /_next/image?url=%2Fsport-acceuil.jpg&w=750&q=75 750w, /_next/image?url=%2Fsport-acceuil.jpg&w=828&q=75 828w, /_next/image?url=%2Fsport-acceuil.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fsport-acceuil.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fsport-acceuil.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fsport-acceuil.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fsport-acceuil.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="absolute inset-0 bg-gradient-to-b from-transparent to-card"
      />
    </div>
    <section
      class="w-full py-10 px-6 text-center"
    >
      <h1
        class="text-4xl md:text-5xl font-bold mb-4"
      >
        Atteins tes objectifs avec 
        <span
          class="text-primary"
        >
          LTcoaching
        </span>
      </h1>
      <p
        class="text-lg md:text-xl text-muted-foreground max-w-2xl mx-auto"
      >
        Coaching personnalisé, suivi nutritionnel, préparation mentale.
        <br />
        Des coachs diplômés à tes côtés, pour atteindre tes objectifs.
      </p>
    </section>
    <section
      class="py-4 px-4 max-w-5xl mx-auto text-center"
    >
      <h2
        class="text-3xl font-bold mb-4 text-primary"
      >
        Qui sommes-nous ?
      </h2>
      <p
        class="text-muted-foreground text-lg max-w-3xl mx-auto"
      >
        Deux Coachs sportifs diplômés avec plus de 7 ans d'expérience.
        <br />
        Nous allons t'aider à transformer ton corps et ton mental.
        <br />
        Que tu veuilles perdre du poids, prendre du muscle, améliorer des qualités physiques specifique, preparation de concours ou tests physiques ou améliorer ta santé global.
        <br />
        Nous t'accompagnerons pas à pas avec des séances sur-mesure personnalisé pour tes objectifs sportif ou santé et un vrai suivi.
      </p>
    </section>
    <section
      class="py-6 px-6 text-center"
    >
      <h2
        class="text-3xl font-bold mb-4 text-primary"
      >
        Prêt à transformer ton corps et ta vie ?
      </h2>
      <p
        class="text-muted-foreground text-lg mb-8"
      >
        Contact nous dès maintenant et construisons ton plan ensemble.
      </p>
      <a
        href="/Contact"
      >
        <button
          class="items-center justify-center gap-2 whitespace-nowrap text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 h-11 mt-6 block mx-auto p-2 font-bold bg-primary text-card rounded-xl hover:scale-105 hover:bg-primary/70"
        >
          Nous contacter
        </button>
      </a>
    </section>
  </main>
</DocumentFragment>
`;
